include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -748,7 +748,7 @@ def BitCastOp: FIRRTLOp<"bitcast", [Pure]> {
   }];
 
   let arguments = (ins FIRRTLBaseType:$input);
```

Comment:
Do we want to allow non-passive input for bitcast?

Reply:
Good question, thanks!  In short, not entirely sure.

It seemed useful and with reasonable expected behavior-- for example, lets you easily get a passive view into a non-passive type (equivalent passive aggregate, or bits).
Offhand not sure if there's anywhere this might be confusing/challenging for our passes/etc., or otherwise bad to allow-- does this seem problematic/inconsistent to you (or for discussion's sake, any arguments to explore re:why this might not be good to support/worth it?).

But I'm not relying on this yet in practice or even conceptually, just am exploring how references to non-passive dataflow points in various scenarios might work and ran into this.

Reply:
I think bitcast is internal only (non-spec), so we can do whatever makes sense.